File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/xray/src/main/java/com/example/xray/DeleteGroup.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.xray;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.xray.XRayClient;
import software.amazon.awssdk.services.xray.model.DeleteGroupRequest;
import software.amazon.awssdk.services.xray.model.XRayException;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class DeleteGroup {

    public static void main(String[] args) {

        final String USAGE = "\n" +
                "Usage: " +
                "   <groupName>\n\n" +
                "Where:\n" +
                "   groupName - the name of the group to delete \n\n";

           if (args.length != 1) {
               System.out.println(USAGE);
               System.exit(1);
         }

        String groupName = args[0];
        Region region = Region.US_EAST_1;
        XRayClient xRayClient = XRayClient.builder()
                .region(region)
                .build();

        deleteSpecificGroup(xRayClient, groupName);
    }

    public static void deleteSpecificGroup(XRayClient xRayClient, String groupName) {

        try {
            DeleteGroupRequest groupRequest = DeleteGroupRequest.builder()
                .groupName(groupName)
                .build();

            xRayClient.deleteGroup(groupRequest);
            System.out.println(groupName + " was deleted!");

        } catch (XRayException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/xray/src/main/kotlin/com/kotlin/xray/DeleteGroup.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.xray

import aws.sdk.kotlin.services.xray.XRayClient
import aws.sdk.kotlin.services.xray.model.DeleteGroupRequest
import kotlin.system.exitProcess

suspend fun main(args:Array<String>) {

    val usage = """
        
        Usage: 
            <groupName>
        
        Where:
            groupName - the name of the group. 
                
        """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
     }

    val groupName = args[0]
    deleteSpecificGroup(groupName)
}

suspend fun deleteSpecificGroup(groupNameVal: String) {

        val groupRequest = DeleteGroupRequest {
            groupName = groupNameVal
        }

        XRayClient { region = "us-east-1" }.use { xRayClient ->
            xRayClient.deleteGroup(groupRequest)
            println("$groupNameVal was deleted!")
        }
    }
